### PR TITLE
Added support for articulations in the Ackermann drive model

### PR DESCRIPTION
**Pull request by Antoni-Robotec**
Added support for articulations in the Ackermann drive model.

### CODE REVIEW

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp [hunk 1]
```diff
@@ -87,27 +89,58 @@ namespace ROS2::VehicleDynamics::Utilities
...
+                articulation = steeringEntityptr->FindComponent<PhysX::ArticulationLinkComponent>();
 
-                if (!hingeComponent)
+                if (!hingeComponent && !articulation)
                 {
                     AZ_Warning(
                         "GetAllSteeringEntitiesData",
                         false,
-                        "Steering entity specified for WheelController in entity %s does not not have HingeJointComponent, ignoring",
+                        "Steering entity specified for WheelController in entity %s does not have a HingeJointComponent nor an "
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp:107]:
```suggestion
                        "Steering entity specified for WheelController in entity %s does not have either a HingeJointComponent or an "
```

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp [hunk 2]
```diff
@@ -152,19 +185,48 @@ namespace ROS2::VehicleDynamics::Utilities
...
                 wheelData.m_wheelEntity = wheel;
-                wheelData.m_hingeJoint = hingeComponent->GetId();
+                if (articulation)
+                {
+                    wheelData.m_wheelJoint = articulation->GetId();
+                    bool hasFreeAxis = Utils::TryGetFreeArticulationAxis(wheelData.m_wheelEntity, wheelData.m_axis);
+                    if (!hasFreeAxis)
+                    {
+                        AZ_Error("VehicleDynamics::Utilities", false, "Articulation wheel has no free axis somehow");
+                    }
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp:224]:
```suggestion
                    const bool hasFreeAxis = Utils::TryGetFreeArticulationAxis(wheelData.m_wheelEntity, wheelData.m_axis);
                    AZ_Error("VehicleDynamics::Utilities", hasFreeAxis, "Articulation wheel has no free axis somehow");
```

File: Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp [hunk 3]
```diff
@@ -87,27 +89,58 @@ namespace ROS2::VehicleDynamics::Utilities
...
-                steeringData.m_hingeJoint = hingeComponent->GetId();
+                steeringData.m_isArticulation = articulation;
+                if (articulation)
+                {
+                    steeringData.m_steeringJoint = articulation->GetId();
+                    bool hasFreeAxis = Utils::TryGetFreeArticulationAxis(steeringData.m_steeringEntity, steeringData.m_axis);
+                    if (!hasFreeAxis)
+                    {
+                        AZ_Error("VehicleDynamics::Utilities", false, "Articulation steering has no free axis somehow");
+                    }
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp:138]:
```suggestion
                    const bool hasFreeAxis = Utils::TryGetFreeArticulationAxis(steeringData.m_steeringEntity, steeringData.m_axis);
                    AZ_Error("VehicleDynamics::Utilities", hasFreeAxis, "Articulation steering has no free axis somehow");
```

**Review by jhanca-robotecai — APPROVED** [review]
The code builds and runs as expected. Only a few nitpicks.

**Review by michalpelka — APPROVED** [review]

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**